includes/class-affiliatewp-allow-own-referrals.php

```diff
@@ -0,0 +1,328 @@
+		 * @since 1.0.3
+		 *
+		 * @return void
+		 */
+		protected function __wakeup() {
+			// Unserializing instances of the class is forbidden
+			_doing_it_wrong( __FUNCTION__, __( 'Cheatin&#8217; huh?', 'affiliatewp-allow-own-referrals' ), '1.0' );
+		}
+
+		/**
+		 * Sets up the class.
+		 *
+		 * @since 1.0.3
+		 */
+		private function __construct() {
+			self::$instance = $this;
+		}
+
+		/**
+		 * Resets the instance of the class.
+		 *
+		 * @since 1.0.3
+		 * @static
+		 */
+		public static function reset() {
+			self::$instance = null;
+		}
+
+		/**
+		 * Sets up plugin constants.
+		 *
+		 * @since 1.0.3
+		 *
+		 * @return void
+		 */
+		private function setup_constants() {
+			// Plugin version.
+			if ( ! defined( 'AFFWP_AOR_VERSION' ) ) {
+				define( 'AFFWP_AOR_VERSION', $this->version );
+			}
+
+			// Plugin Folder Path.
+			if ( ! defined( 'AFFWP_AOR_PLUGIN_DIR' ) ) {
+				define( 'AFFWP_AOR_PLUGIN_DIR', plugin_dir_path( __FILE__ ) );
+			}
+
+			// Plugin Folder URL.
+			if ( ! defined( 'AFFWP_AOR_PLUGIN_URL' ) ) {
+				define( 'AFFWP_AOR_PLUGIN_URL', plugin_dir_url( __FILE__ ) );
+			}
+
+			// Plugin Root File.
+			if ( ! defined( 'AFFWP_AOR_PLUGIN_FILE' ) ) {
+				define( 'AFFWP_AOR_PLUGIN_FILE', __FILE__ );
+			}
+		}
+
+		/**
+		 * Loads the add-on language files.
+		 *
+		 * @since 1.0.3
+		 *
+		 * @return void
+		 */
+		public function load_textdomain() {
+
+			// Set filter for plugin's languages directory.
+			$lang_dir = dirname( plugin_basename( __FILE__ ) ) . '/languages/';
+
+			/**
+			 * Filters the languages directory for the add-on.
+			 *
+			 * @since 1.0.3
+			 *
+			 * @param string $lang_dir Language directory.
+			 */
+			$lang_dir = apply_filters( 'affiliatewp_allow_own_referrals_languages_directory', $lang_dir );
+
+			// Traditional WordPress plugin locale filter..
+			$locale = apply_filters( 'plugin_locale', get_locale(), 'affiliatewp-allow-own-referrals' );
+			$mofile = sprintf( '%1$s-%2$s.mo', 'affiliatewp-allow-own-referrals', $locale );
+
+			// Setup paths to current locale file.
+			$mofile_local  = $lang_dir . $mofile;
+			$mofile_global = WP_LANG_DIR . '/affiliatewp-allow-own-referrals/' . $mofile;
+
+			if ( file_exists( $mofile_global ) ) {
+
+				// Look in global /wp-content/languages/affiliatewp-allow-own-referrals/ folder.
+				load_textdomain( 'affiliatewp-allow-own-referrals', $mofile_global );
+
+			} elseif ( file_exists( $mofile_local ) ) {
+
+				// Look in local /wp-content/plugins/affiliatewp-allow-own-referrals/languages/ folder.
+				load_textdomain( 'affiliatewp-allow-own-referrals', $mofile_local );
+
+			} else {
+
+				// Load the default language files.
+				load_plugin_textdomain( 'affiliatewp-allow-own-referrals', false, $lang_dir );
+
+			}
+		}
+
+		/**
+		 * Includes necessary files.
+		 *
+		 * @since 1.0.3
+		 *
+		 * @return void
+		 */
+		private function includes() {
+			// Bring in the autoloader.
+			require_once dirname( __FILE__ ) . '/lib/autoload.php';
```

Comment:
Can use `__DIR__` here as we're already PHP 5.3 minimum.
```suggestion
			require_once __DIR__ . '/lib/autoload.php';
```